src/main/java/seedu/address/logic/commands/DisinviteCommand.java

```diff
@@ -0,0 +1,150 @@
+            + "involved in expense(s).";
+
+    private final List<String> peopleToDisinvite;
+
+    /**
+     * Creates a DisinviteCommand to disinvite the specified {@code Person(s)}
+     */
+    public DisinviteCommand(List<String> peopleToDisinvite) {
+        requireNonNull(peopleToDisinvite);
+        this.peopleToDisinvite = peopleToDisinvite;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (peopleToDisinvite.size() == 0) {
+            throw new CommandException(MESSAGE_NO_ONE_TO_DISINVITE);
+        }
+
+        if (model.getContext().getType() != ContextType.VIEW_ACTIVITY) {
+            throw new CommandException(MESSAGE_NO_VIEWED_ACTIVITY);
+        }
+
+        Activity activityToDisinviteFrom = model.getContext().getActivity().get();
+        ArrayList<Integer> participantIds = activityToDisinviteFrom.getParticipantIds();
+        model.updateFilteredPersonList(x -> participantIds.contains(x.getPrimaryKey()));
+        List<Person> searchScope = model.getFilteredPersonList();
+
+        List<String> keywords;
+        List<Person> findResult;
+        List<Integer> idsToRemove = new ArrayList<>();
+        List<String> namesToRemove = new ArrayList<>();
+        StringBuilder warningMessage = new StringBuilder();
+        StringBuilder successMessage = new StringBuilder();
+
+        for (String name : peopleToDisinvite) {
+            if (name.trim().equals("")) {
+                warningMessage.append(MESSAGE_INVALID_PERSON + "\n");
+                continue;
+            }
+
+            keywords = Arrays.asList(name.split(" "));
+            NameContainsAllKeywordsPredicate predicate = new NameContainsAllKeywordsPredicate(keywords);
+
+            findResult = searchScope.stream().filter(predicate).collect(Collectors.toList());
+
+            assert findResult != null : "List of people should not be null.";
+
+            if (findResult.size() != 1) { //not in activity or duplicate
+                String warning = String.format(MESSAGE_NON_UNIQUE_SEARCH_RESULT, name);
+                warningMessage.append(warning).append("\n");
+                continue;
+            }
+
+            Person personToDisinvite = findResult.get(0);
+            Integer idOfPersonToDisinvite = personToDisinvite.getPrimaryKey(); // id of person in the activity
+
+            if (idsToRemove.contains(idOfPersonToDisinvite)) { // repeated entry
+                continue;
+            }
+
+            idsToRemove.add(idOfPersonToDisinvite);
+            namesToRemove.add(name);
+        }
+
+        for (int i = 0; i < idsToRemove.size(); i++) {
+            Integer id = idsToRemove.get(i);
+            String name = namesToRemove.get(i);
+            activityToDisinviteFrom.disinvite(id);
+            if (participantIds.contains(id)) { // not removed, has existing expenses
```

Comment:
i think this condition only checks if a participant is in an activity, from what i can tell there's another `participantsActive` array inside activity that needs to be checked

Reply:
alright, like you said i realised the logic is already covered in activity's `disinvite`